### PR TITLE
internal/perms: Fix grant parser bugs

### PR DESCRIPTION
**Pull request by johanbrandhorst**
This fixes two bugs in the grant parser:

1. The parser would accept "," as an output field when parsing from JSON. This would then make it into the canonical string, which would make it look like we had a trailing comma. Disallow the use of a comma as an output field.
2. The parser would accept empty IDs when parsing from JSON. Make this behavior consistent with the text behavior.

### CODE REVIEW

File: internal/perms/grants.go [hunk 1]
```diff
@@ -235,6 +235,9 @@ func (g *Grant) unmarshalJSON(ctx context.Context, data []byte) error {
 			if !ok {
 				return errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("unable to interpret %q element %q as string", "ids", id))
 			}
+			if idStr == "" {
+				return errors.New(ctx, errors.InvalidParameter, op, `segment "ids=" not formatted correctly, missing value`)
```
Comment (jefferai) [inline internal/perms/grants.go:239]: Can this be more specific, as in the error on line 236? "ids" isn't missing a value, some id within it is...

**Review by jefferai — APPROVED** [review]

**Review by jimlambrt — APPROVED** [review]
Ty!

**Review by jimlambrt — APPROVED** [review]
Ty!

File: internal/perms/grants.go [hunk 2]
```diff
@@ -315,9 +329,17 @@ func (g *Grant) unmarshalText(ctx context.Context, grantString string) error {
 		switch kv[0] {
 		case "id":
 			g.id = kv[1]
+			if strings.Contains(g.id, ",") {
+				return errors.New(ctx, errors.InvalidParameter, op, "ID cannot contain a comma, semicolon or equals sign")
```
Comment (jefferai) [inline internal/perms/grants.go:333]: This doesn't match the actual check. Not sure if you intended to check for semicolon/equal or not here.

Comment (johanbrandhorst) [reply internal/perms/grants.go:333]: Good spot, I will update this error message. The parser makes it impossible for an ID to contain a semicolon or equals sign by this point.

**Pull request merged**